platform/ui/src/hooks/useMedia.js

```diff
@@ -17,36 +33,37 @@ import { useState, useEffect } from 'react';
  * @returns
  */
 function useMedia(queries, values, defaultValue) {
-  // Array containing a media query list for each query
-  const mediaQueryLists = queries.map(q => window.matchMedia(q));
+  const memoizedQueryMap = useMemo(() => getMediaQueryMap(queries), [queries]);
+  const memoizedValue = useMemo(
+    () => getValue(memoizedQueryMap, values, defaultValue),
+    [mediaQueryLists, values]
+  );
```

Comment:
So we're trying to be a bit more efficient.
Were we seeing a lot of re-renders? Does this prevent renders unless:

1. A new query is added to the map
2. One or more queries returns a new value (true/false) for matching the current window width

If so, might be worth adding to the JSDoc for `useMedia` -- this is a neat trick